praw/models/reddit/reasons.py

```diff
@@ -0,0 +1,170 @@
+           print(reason)
+
+        """
+        return Reason(self._reddit, self.subreddit, reason_id)
+
+    def __init__(self, subreddit):
+        """Create a SubredditReasons instance.
+
+        :param subreddit: The subreddit whose removal reasons to work with.
+
+        """
+        self.subreddit = subreddit
+        self._reddit = subreddit._reddit
+
+    def __iter__(self):
+        """Return a list of Removal Reasons for the subreddit.
+
+        This method is used to discover all removal reasons for a
+        subreddit:
+
+        .. code-block:: python
+
+           for reason in reddit.subreddit('NAME').reasons:
+               print(reason)
+
+        """
+        response = self.subreddit._reddit.get(
+            API_PATH["removal_reasons_list"].format(subreddit=self.subreddit)
+        )
+        for reason_id, reason_data in response["data"].items():
+            yield Reason(
+                self._reddit, self.subreddit, reason_id, _data=reason_data
+            )
+
+    def add(self, title, message):
+        """Add a removal reason to this subreddit.
+
+        :param title: The title of the removal reason
+        :param message: The message to send the user.
+        :returns: The Reason added.
+
+        The message will be prepended with `Hi u/username,` automatically.
+
+        To add ``'test'`` to the subreddit ``'praw_test'`` try:
+
+        .. code:: python
+
+           reddit.subreddit('praw_test').reason.add('test','This is a test')
+
+        """
+        data = {"title": title, "message": message}
+        url = API_PATH["removal_reasons_list"].format(subreddit=self.subreddit)
+        reason_id = self.subreddit._reddit.post(url, data=data)
+        return Reason(self._reddit, self.subreddit, reason_id)
+
+    def update(self, reason_id, title, message):
+        """Update the removal reason from this subreddit by ``reason_id``.
+
+        :param reason_id: The id of the removal reason
+        :param title: The removal reason's new title (required).
+        :param message: The removal reason's new message (required).
+
+        To update ``'141vv5c16py7d'`` from the subreddit ``'praw_test'`` try:
+
+        .. code:: python
+
+           reddit.subreddit('praw_test').reason.update(
+               '141vv5c16py7d',
+               'New Title',
+               'New message')
+
+        """
+        url = API_PATH["removal_reason"].format(
+            subreddit=self.subreddit, id=reason_id
+        )
+        data = {"title": title, "message": message}
+        self.subreddit._reddit.put(url, data=data)
+
+    def delete(self, reason_id):
+        """Delete a removal reason from this subreddit.
+
+        :param reason_id: The id of the removal reason to remove
+
+        To delete ``'141vv5c16py7d'`` from the subreddit ``'praw_test'`` try:
+
+        .. code:: python
+
+           reddit.subreddit('praw_test').reason.delete('141vv5c16py7d')
```

Comment:
While these example will need to change, please double check to ensure `reason` is pluralized.